src/frontends/lean/pp.cpp

```diff
@@ -1460,7 +1460,7 @@ auto pretty_fn::pp_notation(notation_entry const & entry, buffer<optional<expr>>
                     result e_r   = pp_notation_child(e, token_lbp, a.rbp());
                     curr = tk_fmt + e_r.fmt();
                     if (last)
-                        last_rbp = a.rbp();
+                        last_rbp = std::min(a.rbp(), e_r.rbp());
                     break;
```

Comment:
It would be nice to document why we need `std::min`, and the long discussion we had on Slack.
We discussed other possible fixes, including the "ideal one". I think all this information is important to keep.

Reply:
Forget previous comment, I wrote it before I reviewed the following commit.